### ISSUE TITLE
Conan AutoToolsBuildEnvironment missing `LD_LIBRARY_PATH` and `DYLIB_LIBRARY_PATH`

### ISSUE THREAD

**Issue by ghost** [conan-io/conan#4342]
To help us debug your issue please explain:

- I'm using Conan 1.11.2 on macOS 10.14
- I have a package which depends on library X (boost), dynamically linked
- In my recipe, I defined a requirement on this library and in the config method I set the library option on shared
- The package uses configure/make to generate the package, so I use the AutoToolsBuildEnvironment helper.
- The problem is that AutoToolsBuildEnvironment doesn't set `LD_LIBRARY_PATH` and `DYLIB_LIBRARY_PATH` before running the commands, so the configure step fails as soon as it try to compile and execute something, complaining that it misses the dylib files

```
    requires = "Boost/1.69.0@Boost/stable"
...

    def config(self):
        if self.settings.os == "Macos":
            self.options["Boost"].shared = True
...
    def build(self):
                autotools = AutoToolsBuildEnvironment(self)
                autotools.fpic = True

                args = []
                if not self.options.shared:
                    args += ["--disable-shared"]

                autotools.configure(args=args)
                autotools.make()

```

By replacing the Autotools with `self.run("./configure")`, everything builds fine.
I can also copy the dylibs in the build folder, but I would like to use the helper for consistency with other packages.

---

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

**Comment by SSE4** [comment]
@alessandrodn-cybereason if you run `self.run("./configure")`, how do you get `LD_LIBRARY_PATH`/`DYLD_LIBRARY_PATH` set?

**Comment by alessandrodn-cybereason** [comment]
@SSE4 it's exactly the opposite: when I use `self.run("./configure")` there is no compiler dependency on Boost, so the configure step succeeds.

When I use autoconf, the compiler and linker have the options to link Boost dynamically, so when configure tries to compile and execute something, it requires the Boost dylib in the same folder, unless the variables `LD_LIBRARY_PATH` and `DYLIB_LIBRARY_PATH` are set.

When I use `self.run("./configure")` the compiler and the linker are not aware of the Boost dependency at all.

I would like either to set `LD_LIBRARY_PATH` and `DYLIB_LIBRARY_PATH` for AutoToolsBuildEnvironment or to have it set automatically if it makes sense.

**Comment by SSE4** [comment]
so, it auto-detects boost
I've run into similar issue a couple of times already. I believe `AutoToolsBuildEnvionment` has to run `configure` / `make` commands with `self.run(command, run_environment=True)`, as configure scripts often try to link and run test programs.

**Comment by lasote** [comment]
Probably changing that behavior would be breaking.
Have you tried to use the `RunEnvironment`?

```python
from conans import ConanFile, RunEnvironment

class ExampleConan(ConanFile):
   ...

   def build(self):
      env_build = RunEnvironment(self)
      with tools.environment_append(env_build.vars):
           autotools = AutoToolsBuildEnvironment(self)
           autotools.fpic = True

           args = []
           if not self.options.shared:
               args += ["--disable-shared"]

           autotools.configure(args=args)
           autotools.make()
```

**Comment by leaxdc** [comment]
What's status of this? It's already November....

probably you can fix this by something like this:

`             lib_path_tmpl =  '{}/lib'
             os.environ["LD_LIBRARY_PATH"] = '{}:{}'.format(lib_path_tmpl.format(self.deps_cpp_info['dependency-1'].rootpath),
                                                lib_path_tmpl.format(self.deps_cpp_info['dependency2'].rootpath))`

before env_build.configure() call

**Comment by fulara** [comment]
I am also slightly confused why `LD_LIBRARY_PATH` is not set.
I used myself `RunConfiguration` it has solved very similar issue for me.  
However it seems that `AutoToolsBuildEnvironment` should set in my opinion on **linux**:
* CXX, CC
* LD_LIBRARY_PATH

**Comment by ericLemanissier** [comment]
What is the plan with this issue. I just had yet another failure: https://github.com/bincrafters/conan-libsndfile/runs/392777792#step:5:487 which was solved by using `RunEnvironment()` : https://github.com/bincrafters/conan-libsndfile/commit/3e24808d29f78d2404444cc1b31b236b4f51d9d6 / https://github.com/bincrafters/conan-libsndfile/runs/392792986
`RunEnvironment` is not a proper solution, because it is forbidden in CCI

**Comment by memsharded** [comment]
``VirtualBuildEnvironment``, available in Conan 1.39 (https://docs.conan.io/en/latest/changelog.html#jul-2021), manages these variables now, automatically. Also ``AutoToolsBuildEnvironment`` is going to be removed in 2.0, and is being replaced by ``conan.tools.gnu`` tools like ``AutotoolsToolchain``, so this issue is obsolete now. Closing it, these new tools seems to fullfill the above fine, if something is failing please report a new ticket against them, thanks.
